src/api/cvc4cpp.cpp

```diff
@@ -1873,348 +1906,868 @@ Term Solver::mkFalse(void) const { return d_exprMgr->mkConst<bool>(false); }
+      // until it is substituted away
+    }
+    else if (kind == CONST_RATIONAL)
+    {
+      res = d_exprMgr->mkConst(CVC4::Rational(arg));
+      (void)res.d_expr->getType(true);  /* kick off type checking */
+    }
+    else
+    {
+      res = d_exprMgr->mkConst(CVC4::BitVector(arg));
+      (void)res.d_expr->getType(true);  /* kick off type checking */
+    }
+    return res;
+  }
+  catch (TypeCheckingException& e)
+  {
+    throw CVC4ApiException(e.getMessage());
+  }
+}
+
+Term Solver::mkConst(Kind kind, int32_t arg) const
+{
+  try
+  {
+    CVC4_API_KIND_CHECK_EXPECTED(
+        kind == ABSTRACT_VALUE || kind == CONST_RATIONAL, kind)
+        << "ABSTRACT_VALUE or CONST_RATIONAL";
+    Term res;
+    if (kind == ABSTRACT_VALUE)
+    {
+      res = d_exprMgr->mkConst(CVC4::AbstractValue(Integer(arg)));
+      // do not call getType(), for abstract values, type can not be computed
+      // until it is substituted away
+    }
+    else
+    {
+      res = d_exprMgr->mkConst(CVC4::Rational(arg));
+      (void)res.d_expr->getType(true);  /* kick off type checking */
+    }
+    return res;
+  }
+  catch (TypeCheckingException& e)
+  {
+    throw CVC4ApiException(e.getMessage());
+  }
+}
+
+Term Solver::mkConst(Kind kind, int64_t arg) const
+{
+  try
+  {
+    CVC4_API_KIND_CHECK_EXPECTED(
+        kind == ABSTRACT_VALUE || kind == CONST_RATIONAL, kind)
+        << "ABSTRACT_VALUE or CONST_RATIONAL";
+    Term res;
+    if (kind == ABSTRACT_VALUE)
+    {
+      res = d_exprMgr->mkConst(CVC4::AbstractValue(Integer(arg)));
+      // do not call getType(), for abstract values, type can not be computed
+      // until it is substituted away
+    }
+    else
+    {
+      res = d_exprMgr->mkConst(CVC4::Rational(arg));
+      (void)res.d_expr->getType(true);  /* kick off type checking */
+    }
+    return res;
+  }
+  catch (TypeCheckingException& e)
+  {
+    throw CVC4ApiException(e.getMessage());
+  }
+}
+
+Term Solver::mkConst(Kind kind, uint64_t arg) const
+{
+  try
+  {
+    CVC4_API_KIND_CHECK_EXPECTED(
+        kind == ABSTRACT_VALUE || kind == CONST_RATIONAL, kind)
+        << "ABSTRACT_VALUE or CONST_RATIONAL";
+    Term res;
+    if (kind == ABSTRACT_VALUE)
+    {
+      res = d_exprMgr->mkConst(CVC4::AbstractValue(Integer(arg)));
+      // do not call getType(), for abstract values, type can not be computed
+      // until it is substituted away
+    }
+    else
+    {
+      res = d_exprMgr->mkConst(CVC4::Rational(arg));
+      (void)res.d_expr->getType(true);  /* kick off type checking */
+    }
+    return res;
+  }
+  catch (TypeCheckingException& e)
+  {
+    throw CVC4ApiException(e.getMessage());
   }
-  return d_exprMgr->mkConst(CVC4::Rational(arg));
 }
 
 Term Solver::mkConst(Kind kind, uint32_t arg1, uint32_t arg2) const
 {
-  CVC4_API_KIND_CHECK_EXPECTED(kind == CONST_RATIONAL, kind)
-      << "CONST_RATIONAL";
-  return d_exprMgr->mkConst(CVC4::Rational(arg1, arg2));
+  try
```

Comment:
Again template helper for `uint32_t`, `int32_t`, `int64_t`, and `uint64_t`?

Reply:
Ditto